processor/src/lib.rs

```diff
@@ -584,7 +587,9 @@ where
                 self.host.borrow_mut().set_advice(self, *injector)?;
             },
             Decorator::Debug(options) => {
-                self.host.borrow_mut().on_debug(self, options)?;
+                if self.enable_debugging {
```

Comment:
Similar to the above comment, we could get this info from the decoder. So, this could become `if self.decoder.in_debug_mode()` (similar how we have it on line 595 below).